### PR TITLE
[SPARK-49757][SQL] Support IDENTIFIER expression in SET CATALOG statement 

### PR DESCRIPTION
**Pull request by mikhailnik-db**
### What changes were proposed in this pull request?

This pr adds possibility to use `IDENTIFIER(...)` for a catalog name in `SET CATALOG` statement.
For instance `SET CATALOG IDENTIFIER('test')` now works the same as `SET CATALOG test`

### Why are the changes needed?

1. Consistency of API. It can be confusing for user that he can use IDENTIFIER in some contexts but cannot for catalogs.
2. Parametrization. It allows user to write `SET CATALOG IDENTIFIER(:user_data)` and doesn't worry about SQL injections.

### Does this PR introduce _any_ user-facing change?

Yes, now `SET CATALOG IDENTIFIER(...)` works. It can be used with any string expressions and parametrization.
But multipart identifiers (like `IDENTIFIER('database.table')`) are banned and will rise ParseException with `INVALID_SQL_SYNTAX.MULTI_PART_NAME`. This restriction always has been on grammar level, but now user can try to bind such identifier via parameters.

### How was this patch tested?

Unit tests with several new covering new behavior.

### Was this patch authored or co-authored using generative AI tooling?

Yes, some code suggestions
Generated-by: GitHub Copilot

### CODE REVIEW

File: common/utils/src/main/resources/error/error-conditions.json [hunk 1]
```diff
@@ -3021,6 +3021,11 @@
...
         ]
       },
+      "MULTI_PART_CATALOG_NAME" : {
+        "message" : [
+          "<statement> with multiple part catalog name(<catalogName>) is not allowed."
+        ]
+      },
       "MULTI_PART_NAME" : {
         "message" : [
           "<statement> with multiple part function name(<funcName>) is not allowed."
```
Comment (mikhailnik-db) [inline common/utils/src/main/resources/error/error-conditions.json:3031]: I'm not sure if it's better to leave this two errors separately or generalized MULTI_PART_NAME to match all use cases

Comment (cloud-fan) [reply common/utils/src/main/resources/error/error-conditions.json:3031]: I'd like to have a generalized error
```
"<statement> with multiple part name(<name>) is not allowed."
```

Comment (mikhailnik-db) [reply common/utils/src/main/resources/error/error-conditions.json]: done

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
The test failure is unrelated, thanks, merging to master!

File: sql/api/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 2]
```diff
@@ -621,9 +621,8 @@ private[sql] object QueryParsingErrors extends DataTypeErrorsBase {
   def unsupportedFunctionNameError(funcName: Seq[String], ctx: ParserRuleContext): Throwable = {
     new ParseException(
       errorClass = "INVALID_SQL_SYNTAX.MULTI_PART_NAME",
-      messageParameters = Map(
-        "statement" -> toSQLStmt("CREATE TEMPORARY FUNCTION"),
-        "funcName" -> toSQLId(funcName)),
+      messageParameters =
+        Map("statement" -> toSQLStmt("CREATE TEMPORARY FUNCTION"), "name" -> toSQLId(funcName)),
```
Comment (uros-db) [inline sql/api/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:625]: I see a lot of these changes (`funcName` to `name`) in this PR

is this related to the core changes? (supporting ident in SET CATALOG)

**Review by uros-db — COMMENTED** [review]
do we want to support `USE CATALOG` with ident as well?

File: sql/api/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 2 above)
Comment (cloud-fan) [reply sql/api/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:625]: Yes, it's not always function name now, but can also be catalog name.
